ambry-router/src/test/java/com.github.ambry.router/GetManagerTest.java

```diff
@@ -0,0 +1,172 @@
+  private NonBlockingRouter router;
+  private int chunkSize;
+  private int requestParallelism;
+  private int successTarget;
+  // Request params;
+  private BlobProperties putBlobProperties;
+  private byte[] putUserMetadata;
+  private byte[] putContent;
+  private ReadableStreamChannel putChannel;
+
+  private static final int MAX_PORTS_PLAIN_TEXT = 3;
+  private static final int MAX_PORTS_SSL = 3;
+  private static final int CHECKOUT_TIMEOUT_MS = 1000;
+
+  /**
+   * Pre-initialization common to all tests.
+   */
+  public GetManagerTest()
+      throws Exception {
+    // random chunkSize in the range [1, 1 MB]
+    chunkSize = random.nextInt(1024 * 1024) + 1;
+    requestParallelism = 3;
+    successTarget = 2;
+    mockSelectorState.set(MockSelectorState.Good);
+    mockClusterMap = new MockClusterMap();
+    mockServerLayout = new MockServerLayout(mockClusterMap);
+  }
+
+  /**
+   * Every test in this class should leave the router closed in the end. Some tests do additional checks after
+   * closing the router. This is just a guard to ensure that the tests are not broken (which helped when developing
+   * these tests).
+   */
+  @After
+  public void postCheck() {
+    Assert.assertFalse("Router should be closed at the end of each test", router.isOpen());
+    Assert.assertEquals("Router operations count must be zero", 0, NonBlockingRouter.getOperationsCount());
+  }
+
+  /**
+   * Tests get blob info of simple blobs
+   * @throws Exception
+   */
+  @Test
+  public void testSimpleBlobGetBlobInfoSuccess()
+      throws Exception {
+    router = getNonBlockingRouter();
+    setOperationParams(chunkSize);
+    String blobId = router.putBlob(putBlobProperties, putUserMetadata, putChannel).get();
+    BlobInfo blobInfo = router.getBlobInfo(blobId).get();
+    Assert.assertTrue("Blob properties should match",
+        RouterTestHelpers.haveEquivalentFields(putBlobProperties, blobInfo.getBlobProperties()));
+    Assert.assertArrayEquals("User metadata should match", putUserMetadata, blobInfo.getUserMetadata());
+    router.close();
+  }
+
+  /**
+   * Tests get blob info of composite blobs
+   * @throws Exception
+   */
+  @Test
+  public void testCompositeBlobGetBlobInfoSuccess()
+      throws Exception {
+    router = getNonBlockingRouter();
+    setOperationParams(chunkSize * 6 + 11);
+    String blobId = router.putBlob(putBlobProperties, putUserMetadata, putChannel).get();
+    BlobInfo blobInfo = router.getBlobInfo(blobId).get();
+    Assert.assertTrue("Blob properties should match",
+        RouterTestHelpers.haveEquivalentFields(putBlobProperties, blobInfo.getBlobProperties()));
+    Assert.assertArrayEquals("User metadata should match", putUserMetadata, blobInfo.getUserMetadata());
+    router.close();
+  }
+
+  /**
+   * Tests the failure case where poll throws and closes the router. This also tests the case where the GetManager
```

Comment:
tests the case *where* the `GetManager` gets closed, not *that* it is closed specifically. Basically, ensures that the `GetManager#close()` is called and completes existing operations with the `RouterClosed` error, by checking exactly that.